ads/inabox/inabox-messaging-host.js

```diff
@@ -224,15 +224,26 @@ export class InaboxMessagingHost {
       return this.iframeMap_[sentinel];
     }
 
-    // Walk up on the window tree until find host's direct child window
-    while (source.parent !== this.win_ && source !== this.win_.top) {
-      source = source.parent;
-    }
-
-    // Find the iframe element that hosts the message source window
+    // Verify that the source of the postMessage corresponds to one of
+    // the iframes that was registered.
     for (let i = 0; i < this.iframes_.length; i++) {
       const iframe = this.iframes_[i];
       if (iframe.contentWindow === source) {
+	  // Walk up on the window tree and verify that there is no cross-domain
+	  // frame in between the source of the postMessage and the host doc.
+	  // If there is, the host doc will not be able to accurately measure
+	  // the creative's positions in the page, so return null.
+	  // Limit to 10 iterations.
+	  let tempWin = source.parent;
+	  for (let i = 0; i < 10; i++) {
```

Comment:
You can move the if break statement directly into the for loop declaration, I also think we should make sure tempWin exists:

for (let i = 0; i < 10 && tempWin && tempWin != this.win_ && tempWin != this.win_.top) {

Reply:
done

Reply:
BTW if you want to be really fancy you can include the declaration of tempWin:

for (let i = 0, tempWin = source.parent; i < 10 && tempWin && tempWin != this.win_ && tempWin != this.win_.top; i++, tempWin = source.parent) {
   if (!canInspectWindow(tempWin)) {
      return null;
   }
}

Now that I think of it, your loop is not properly returning null when we hit the more than 10 case, so really this should be:

for (let i = 0, tempWin = source.parent; tempWin && tempWin != this.win_ && tempWin != this.win_.top; i++, tempWin = source.parent) {
   if (i == 10 || !canInspectWindow(tempWin)) return null;
}

Reply:
done